### ISSUE TITLE
Rescuing StandardError can lead to some really weird bugs!

### ISSUE THREAD

**Issue by drinks** [customink/vertex_client#47]
@WesKetch and I just worked through a strange one:

We don't configure the circuit in test, with the hope that the circuit will never open and errors will make their way up the stack to us to help us debug. We also use some [just-in-time cassette swapping code](https://github.com/customink/stores/blob/master/test/support/vcr_helper.rb#L21-L29) to keep vertex calls segmented because they are likely to introduce churn in a cassette suite.

With that setup, we ran into a spot where we hadn't remembered to insert a cassette. This should have errored because VCR didn't find the expected cassette, but it errored first when we tried to get the name of the cassette to add to the vertex_client folder. But the test passed, because VertexClient rescued the NoMethodError that was raised(!)

However, all future tests needing to make http calls failed, because the cassette stack got out of sync and nothing else could find the name of the cassette that it should insert. 😱 

TL;DR 
Is anyone aware of a sane set of explicit exceptions we should rescue [here](https://github.com/customink/vertex_client/blob/059efc43562a8e0bc769e2713b96f596222f1097/lib/vertex_client/connection.rb#L49)? Any Savon::Error for sure, but there are likely others?
